include/openthread/cli.h

```diff
@@ -148,6 +148,15 @@ void otCliAppendResult(otError aError);
  */
 void otCliPlatLogv(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, va_list aArgs);
 
+#if OPENTHREAD_CONFIG_CLI_VENDOR_COMMANDS_ENABLE && OPENTHREAD_CONFIG_CLI_MAX_USER_CMD_ENTRIES > 1
```

Comment:
We cannot and should not use any of `OPENTHREAD_CONFIG` macros in any of public or private OT headers (like `cli.h`) - since the header may be included in other projects where these constants are not available.

The approach is we document that API requires certain `OPENTHREAD_CONFIG`. And provide implementation of it under the config.

Reply:
> We cannot and should not use any of `OPENTHREAD_CONFIG` macros in any of public or private OT headers (like `cli.h`) - since the header may be included in other projects where these definitions are not available/provided
> 
> The approach is that we document that API requires certain `OPENTHREAD_CONFIG`. And provide the implementation of it under the config only.

What if I just remove the #if macro check?

Reply:
Yes. This is the way